test/powershell/Host/ConsoleHost.Tests.ps1

```diff
@@ -711,8 +711,52 @@ namespace StackTest {
             $longPipeName = [string]::new("A", 200)
 
             "`$pid" | & $powershell -CustomPipeName $longPipeName -c -
-            # 64 is the ExitCode for BadCommandLineParameter
-            $LASTEXITCODE | Should -Be 64
+            $LASTEXITCODE | Should -Be $ExitCodeBadCommandLineParameter
+        }
+    }
+
+    Context "ApartmentState tests" -Tag Slow {
+
+        It "Default apartment state is STA" -Skip:(!$IsWindows) {
+            [System.Threading.Thread]::CurrentThread.ApartmentState | Should -BeExactly "STA"
+        }
+
+        It "Should be able to set apartment state to: <apartment>" -Skip:(!$IsWindows) -TestCases @(
+            @{ apartment = "STA"; switch = "-sta" }
+            @{ apartment = "MTA"; switch = "-mta" }
+        ) {
+            param ($apartment, $switch)
+
+            & $powershell $switch -noprofile -command "[System.Threading.Thread]::CurrentThread.ApartmentState" | Should -BeExactly $apartment
```

Comment:
```suggestion
            & $powershell $switch -noprofile -command "[System.Threading.Thread]::CurrentThread.GetApartmentState()" | Should -BeExactly $apartment
```
Should we check both main thread and runspace as mentioned above?

Reply:
Will add seperate test for new runspace check

Reply:
On WinPS, a new runspace is MTA, so we're keeping that to be consistent